hearts/game/hearts.py

```diff
@@ -227,12 +239,35 @@ def set_turn_counter(self):
                 self.turn_counter = index
 
     def is_valid_pass_for_player(self, player, cards):
-        # [Card] --> Bool
-        return len(cards) == 3 and all(card in self.hands[player] for card in cards)
+        # [Card] --> (Bool, string)
+        is_valid = True
+        string_of_cards = ', '.join(str(card) for card in cards)
+        error_string = None
+
+        if len(cards) != 3:
+            error_string = message(INVALID_PASS, NOT_THREE).format(string_of_cards)
+            is_valid = False
+        elif not all(card in self.hands[player] for card in cards):
+            illegal_card = [str(card) for card in cards if card not in self.hands[player]][0]
+            error_string = message(INVALID_PASS, NOT_IN_HAND).format(string_of_cards, illegal_card)
+            is_valid = False
+        return (is_valid, error_string)
+
+    def add_to_pass_selections(self, player, cards):
```

Comment:
@j2kun I will probably need to do a refactoring of the passing feature (later on) once we decide how passing works through the api.